internal/receiver/smartagentreceiver/config.go

```diff
@@ -145,6 +140,29 @@ func mergeConfigs(componentViperSection *viper.Viper, intoCfg interface{}) error
 	return nil
 }
 
+type viperAllSettings map[string]interface{}
+
+func (allSettings viperAllSettings) getPointerToStringSliceFromYaml(key string, errToReturn error) (*[]string, error) {
```

Comment:
Why not just return `[]string` instead of `*[]string`?

Reply:
We need to know if it was unset for default behavior

Reply:
We can check for `nil` value on `[]string` too, right?

Reply:
Oh definitely.  My reasoning was muddled about nil slices not being possible in viper unmarshalling and it led to a bug.  thanks!